crates/uv/tests/it/common/mod.rs

```diff
@@ -507,7 +507,20 @@ impl TestContext {
             for (version, executable) in &python_versions {
                 let parent = python_dir.child(version.to_string());
                 parent.create_dir_all().unwrap();
-                parent.child("python3").symlink_to_file(executable).unwrap();
+                // Ensure the link points to the canonicalized patch version path if
+                // the `python-patch` feature is enabled
+                if cfg!(feature = "python-patch") {
```

Comment:
I worry a little about making this conditional, because this feature is intended for marking specific tests, not for changing the behavior of the test suite as a whole.

Reply:
Interesting. It looks as though the tests pass without this when upgrade is in preview mode. I've removed this for now and will have to revisit when stabilizing transparent upgrades.